### PR TITLE
EP-1 first commit

### PR DESCRIPTION
**Pull request by NotTerrae**

### CODE REVIEW

File: src/components/EmojiContainer/EmojiContainer.module.css [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+.categoryIndividual {
+  margin-bottom: 30px;
+}
+
+.emojis {
+  display: grid;
+  grid-gap: 10px;
+  grid-template-columns: repeat(9, 30px);
+  grid-template-rows: repeat(auto, 30px);
+  /* display: flex;
```
Comment (catalinracz) [inline src/components/EmojiContainer/EmojiContainer.module.css:81]: remove if not used

File: src/components/Emojis/Emojis.module.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/* .categories {
```
Comment (catalinracz) [inline src/components/Emojis/Emojis.module.css:1]: remove if not used

File: src/components/Emojis/Emojis.module.css [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+.categoryIndividual {
+  margin-bottom: 30px;
+}
+
+.emojis {
+  display: grid;
+  grid-gap: 10px;
+  grid-template-columns: repeat(9, 30px);
+  grid-template-rows: repeat(auto, 30px);
+  /* display: flex;
```
Comment (catalinracz) [inline src/components/Emojis/Emojis.module.css:39]: remove if not used

File: src/components/EmojiCategoryBar/EmojiCategoryBar.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import classes from './EmojiCategoryBar.module.css';
+
+const EmojiCategoryBar = ({ content, onCategory }) => {
+  return (
+    <div className={classes.categories}>
+      <div onClick={() => onCategory('')} className={classes.categoryIcon}>
+        💯
```
Comment (andreipop02) [inline src/components/EmojiCategoryBar/EmojiCategoryBar.js:7]: I'm not sure about having this character here, it's making the code look a bit weird. Maybe you can create a strings.js file and import it from there:

Example:
strings.js

export const strings = {
sign100: "💯"
}

and then in the screen: 

import strings from 'path/strings.js'
.
.
.
div onClick={() => onCategory('')} className={classes.categoryIcon}
        {strings.sign100}

File: src/components/EmojiCategoryBar/EmojiCategoryBar.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      </div>
+      {Object.keys(content).map(group => {
+        return (
+          <div
+            className={classes.categoryIcon}
+            key={Math.random()}
```
Comment (andreipop02) [inline src/components/EmojiCategoryBar/EmojiCategoryBar.js:13]: same as above, try to use another key (being random might result in the same key, even if it's a very small chance)

File: src/components/EmojiContainer/EmojiContainer.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    setSearch,
+    filterBySearch,
+    hasFilteredCategoty,
+    scrollToTop,
+  } = useContext(ContainerContext);
+
+  return (
+    <>
+      <div className={classes.container}>
+        <div id="top"></div>
```
Comment (andreipop02) [inline src/components/EmojiContainer/EmojiContainer.js:26]: we can make this a self-closing tag if we dont wrap anything with it (better-looking)

**Review by catalinracz — CHANGES_REQUESTED** [review]

File: src/components/Emojis/Emojis.module.css [hunk 7]
```diff
@@ -0,0 +1,43 @@
+/* .categories {
+  display: flex;
+  justify-content: space-around;
+  margin: 0 20px 10px 20px;
+  border: 1px solid grey;
+  border-radius: 5px;
+  padding: 8px;
+  box-shadow: 2px 2px 10px rgb(199, 199, 199);
+} */
```
Comment (andreipop02) [inline src/components/Emojis/Emojis.module.css:9]: do we need this code? If not, please remove unused code

File: src/components/Emojis/EmojisByCategory.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+}) => {
+  return (
+    <div className={classes.categoryContainer}>
+      {Object.keys(content)
+        .filter(group => group === category)
+        .map(group => {
+          const emojiInGroup = content[group];
+          return (
+            onNonEmpty(emojiInGroup) && (
+              <div className={classes.categoryIndividual} key={Math.random()}>
```
Comment (andreipop02) [inline src/components/Emojis/EmojisByCategory.js:18]: same as other keys

File: src/index.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import { AppContextProvider } from './context/app-context';
+import { ContainerContextProvider } from './context/container-context';
+import { ChatContextProvider } from './context/chat-context';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <ChatContextProvider>
+    <ContainerContextProvider>
+      <AppContextProvider>
+        {' '}
```
Comment (andreipop02) [inline src/index.js:13]: we can remove this, it's from lint I think, we dont need it

**Review by andreipop02 — CHANGES_REQUESTED** [review]

**Pull request merged**